File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.py
import sys
N, K = map(int, sys.stdin.readline().split(" "))
circle = [i for i in range(1, N + 1)]
answer = []
index = 0
while(len(circle) > 0):
    index = (index + K - 1) % len(circle)
    answer.append(str(circle.pop(index)))
print("<", ", ".join(answer), ">", sep="")

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.js
// 1158 요세푸스 순열
function solution(input) {
  N = input[0];
  K = input[1];
  const people = [];
  const answer = [];
  let index = 0;
  for (let i = 1; i <= N; i++) {
    people.push(i);
  }
  while (people.length > 0) {
    index = (K - 1 + index) % people.length;
    answer.push(...people.splice(index, 1));
  }
  console.log(`<${answer.join(", ")}>`);
}

const fs = require("fs");
const file = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(file).toString().split(" ");
solution(input);
